ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -869,9 +870,23 @@ public void updateDemographicsRecord(List<Map<String, Object>> updatedRows) thro
             }
         }
 
-        ti.getUpdateService().updateRows(getUser(), getContainer(), newRows, keyRows, null, getExtraContext());
-
-        EHRDemographicsService.get().getAnimals(getContainer(), ids);
+        DbScope scope = ti.getSchema().getScope();
+        try (DbScope.Transaction transaction = scope.ensureTransaction())
+        {
+            transaction.addCommitTask(() ->
+            {
+                try
+                {
+                    ti.getUpdateService().updateRows(getUser(), getContainer(), newRows, keyRows, null, getExtraContext());
+                    EHRDemographicsService.get().getAnimals(getContainer(), ids);
+                }
+                catch (InvalidKeyException | BatchValidationException | QueryUpdateServiceException | SQLException e)
+                {
+                    throw new RuntimeException(e);
+                }
+            }, DbScope.CommitTaskOption.POSTCOMMIT);
```

Comment:
This should be PRECOMMIT. It'll run inside the transaction itself, instead of in a separate transaction. It shouldn't impact the trigger script's timing.

Reply:
done